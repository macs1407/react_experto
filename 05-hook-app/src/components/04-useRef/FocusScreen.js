import React, { useRef } from 'react';
import '../02-useEfect/efects.css';

export const FocusScreen = () => {

    // Sirve para tener referencia o seguimiento a un objeto
    const inputRef = useRef()

    const handleKey = ()=>{
        inputRef.current.select();
        console.log(inputRef);
    }

    return (
        <div>
            <h1>Focus</h1>
            <hr />
            <input 
                ref={inputRef}
                className="form-control" 
                placeholder="nombre"
            /><br />
            <button className="btn btn-primary" onClick={handleKey}>Focus</button>
        </div>
    )
}
