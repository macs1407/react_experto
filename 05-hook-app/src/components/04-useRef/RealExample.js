import React, { useState } from 'react';
import '../02-useEfect/efects.css';
import { MultipleCustomHooks } from '../03-examples/MultipleCustomHooks';

export const RealExample = () => {

    const [show, setShow] = useState(false)

    return (
        <div>
            <h1>Real example ref</h1>
            <hr />
            {
                show && (<MultipleCustomHooks />)
            }            
            <button 
                className="btn btn-primary mt-5"
                onClick={ ()=>{setShow(!show)} }
                >
                    Mostrar/Ocultar
            </button>
        </div>
    )
}
