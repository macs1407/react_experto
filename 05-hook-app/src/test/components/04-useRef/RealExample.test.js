import React from 'react';
import '@testing-library/jest-dom';
import { RealExample } from "../../../components/04-useRef/RealExample";
import {shallow} from 'enzyme';

describe('Pruebas con RealExample', () => {
    const wraper = shallow(<RealExample />);

    test('Debe de levantar RealExample ', () => {
        
        expect(wraper).toMatchSnapshot();
    });

    test('Debe de mostrar MultipleCustomHooks ', () => {
        // simular el click
        wraper.find('button').simulate('click');
        expect(wraper.find('MultipleCustomHooks').exists()).toBe(true);
    });
})
