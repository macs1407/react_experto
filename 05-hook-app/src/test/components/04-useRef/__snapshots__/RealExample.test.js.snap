// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas con RealExample Debe de levantar RealExample  1`] = `
<div>
  <h1>
    Real example ref
  </h1>
  <hr />
  <button
    className="btn btn-primary mt-5"
    onClick={[Function]}
  >
    Mostrar/Ocultar
  </button>
</div>
`;
